### PR TITLE
Fix basic auth with encoded characters (#3815)

### PR DESCRIPTION
**Pull request by ZIRAKrezovic**
Fixes: #3815

### CODE REVIEW

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hello @ZIRAKrezovic, thanks for creating the PR. Please add a test. Please add your full name and surname with `@author` tag to the javadoc of any classes you've modified.

**Comment by ZIRAKrezovic** [comment]
Hi @OlgaMaciaszek, I have added a test. Although I'm not sure if it's the right approach, or if it can be done with unit tests without a server, it tests what this fix is supposed to fix.

**Comment by ZIRAKrezovic** [comment]
I have found out how Spring Webflux developers do the tests and updated my test accordingly.

https://github.com/spring-projects/spring-framework/blob/6.1.x/spring-webflux/src/test/java/org/springframework/web/reactive/function/client/DefaultWebClientTests.java#L128

File: spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/http/WebClientTransportClientFactoryTest.java [hunk 1]
```diff
@@ -16,23 +16,57 @@
 
 package org.springframework.cloud.netflix.eureka.http;
 
+import java.time.Duration;
+
 import com.netflix.discovery.shared.resolver.DefaultEndpoint;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
```
Comment (OlgaMaciaszek) [inline spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/http/WebClientTransportClientFactoryTest.java:25]: Please update the year in the license comment.

File: spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/http/WebClientTransportClientFactoryTest.java [hunk 2]
```diff
@@ -16,23 +16,57 @@
...
+
 	private WebClientTransportClientFactory transportClientFatory;
 
 	@BeforeEach
 	void setup() {
-		transportClientFatory = new WebClientTransportClientFactory(WebClient::builder);
+		ClientResponse mockResponse = mock();
+		when(mockResponse.statusCode()).thenReturn(HttpStatus.OK);
+		when(mockResponse.bodyToMono(Void.class)).thenReturn(Mono.empty());
+		given(exchangeFunction.exchange(captor.capture())).willReturn(Mono.just(mockResponse));
```
Comment (OlgaMaciaszek) [inline spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/http/WebClientTransportClientFactoryTest.java:66]: It's a good approach. On a higher level, you could possibly also use `okhttp3.mockwebserver.MockWebServer` to record and verify the response, but the way you've mocked it is good.

File: spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/http/WebClientTransportClientFactoryTest.java [hunk 3]
```diff
@@ -16,23 +16,57 @@
...
+
+		transportClientFatory = new WebClientTransportClientFactory(
+				() -> WebClient.builder().exchangeFunction(this.exchangeFunction));
```
Comment (OlgaMaciaszek) [inline spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/http/WebClientTransportClientFactoryTest.java:69]: Please remove unnecessary `this` keyword.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Thanks @ZIRAKrezovic. Looks good. Have just added comments about some cosmetic issues. Please fix.

**Review by OlgaMaciaszek — APPROVED** [review]
Thanks @ZIRAKrezovic. LGTM.

**Pull request merged**